docs/diagnostic-logs.md

```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
+
+**Details of the message flow**
+
+For the direction **from client to server via SignalR service**, SignalR service will **only** consider the invocation that is originated from diagnostic client, that is, the message generated directly in diagnostic client, or service message generated due to the invocation of diagnostic client indirectly. 
+
+The tracing ID will be generated in SignalR service once the message arrives at SignalR service in **Path 1**. SignalR service will generate a log `Received a message <MessageTracingId> from client connection <ConnectionId>.` for each message in diagnostic client. Once the message leaves from the SignalR to server, SignalR service will generate a log `Sent a message <MessageTracingId> to server connection <ConnectionId> successfully.` If you see these 2 logs, you can be sure that the message passes through SignalR service successfully. 
+
+> Due to the limitation of ASP.NET Core SignalR, the message comes from client doesn't contains any message level ID. But ASP.NET SignalR generate *invocation ID* for each message, you can use it to map with the tracing ID.
+
+Then the message carries the tracing ID Server in **Path 2**. Server will generate a log `Received message <messagetracingId> from client connection <connectionId>` once the message arrives.
+
+<span id="message-flow-detail-for-path3"></span>
+Once the message invokes the hub method in server, a new service message will be generated with a *new tracing ID*. Once the service message is generated, server will generate a log in template `Start to broadcast/send message <MessageTracingId> ...`, the actual log will be based on your scenario. Then the message will be delivered to SignalR service in **Path 3**, once the service message leaves from server, a log called `Succeeded to send message <MessageTracingId>` will be generated. 
+
+> Due to the limitation of SignalR, message tracing ID can't be passed through SignalR hub. 
+
+Once the service message arrives at SignalR service, a log called `Received a <MessageType> message <MessageTracingId> from server connection <ConnectionId>.` will be generated. Then SignalR service processes the service message and deliver to the target client(s). Once the message is sent to client(s) in **Path 4**, log `Sent a message <MessageTracingId> to client connection <ConnectionId> successfully.` will be generated.
+
+In summary, the message log will be generated when message goes in and out the SignalR service and server. You can use these logs to validate whether the message is lost in these components or not.
+
+Below is a typical message loss issue.
+
+###### A user fails to receive messages in a group
+
+The typical story in this issue is that the user joins a group **after** sending a group message.
```

Comment:
Explain this problem a little bit more (for example, a sample code that customer didn't await the group join before send the message to group).

Reply:
added

Reply:
updated  for all the comments at new commit https://github.com/Azure/azure-signalr/pull/1047/commits/60b3edeff9a95bbde78b4f1a06ab19e7fd66c364 , please take a look